iothub/device/tests/Authentication/ClientAuthenticationWithTokenRefreshTests.cs

```diff
@@ -140,109 +140,144 @@ public async Task ClientAuthenticationWithTokenRefresh_GetTokenAsync_NewTtl_Ok()
 
-            Assert.IsNull(iotHubConnectionCredentials.SharedAccessSignature);
-            Assert.AreEqual(TestDeviceId, iotHubConnectionCredentials.DeviceId);
-
-            Assert.IsNotNull(token1);
-            Assert.IsNotNull(token2);
-            Assert.AreEqual(token1, token2.TokenValue);
+            // assert
+            iotHubConnectionCredentials.SharedAccessSignature.Should().BeNull();
+            TestDeviceId.Should().Be(iotHubConnectionCredentials.DeviceId);
+            token1.Should().NotBeNull();
+            token1.Should().NotBeNull();
+            token2.TokenValue.Should().Be(token1);
         }
 
         [TestMethod]
         public async Task ClientAuthenticationWithSakRefresh_WithModule_SharedAccessKeyConnectionString_HasRefresher()
         {
+            // arrange and act
             IConnectionCredentials iotHubConnectionCredentials = new IotHubConnectionCredentials(
                 new ClientAuthenticationWithSharedAccessKeyRefresh(
                     sharedAccessKey: TestSharedAccessKey,
                     deviceId: TestDeviceId,
                     moduleId: TestModuleId),
                 TestIotHubName);
 
-            Assert.IsNotNull(iotHubConnectionCredentials.SasTokenRefresher);
-            Assert.IsInstanceOfType(iotHubConnectionCredentials.SasTokenRefresher, typeof(ClientAuthenticationWithSharedAccessKeyRefresh));
+            // assert
+            iotHubConnectionCredentials.SasTokenRefresher.Should().NotBeNull();
+            iotHubConnectionCredentials.SasTokenRefresher.Should().BeOfType<ClientAuthenticationWithSharedAccessKeyRefresh>();
 
+            // act
             var cbsAuth = new AmqpIotCbsTokenProvider(iotHubConnectionCredentials);
-
             string token1 = await iotHubConnectionCredentials.GetPasswordAsync().ConfigureAwait(false);
             CbsToken token2 = await cbsAuth.GetTokenAsync(new Uri("amqp://" + TestIotHubName), "testAppliesTo", null).ConfigureAwait(false);
 
-            Assert.IsNull(iotHubConnectionCredentials.SharedAccessSignature);
-            Assert.AreEqual(TestDeviceId, iotHubConnectionCredentials.DeviceId);
+            // assert
+            iotHubConnectionCredentials.SharedAccessSignature.Should().BeNull();
+            iotHubConnectionCredentials.DeviceId.Should().Be(TestDeviceId);
+            token1.Should().NotBeNull();
+            token2.Should().NotBeNull();
+            token2.TokenValue.Should().Be(token1);
+        }
 
-            Assert.IsNotNull(token1);
-            Assert.IsNotNull(token2);
-            Assert.AreEqual(token1, token2.TokenValue);
+        [TestMethod]
+        public void ClientAuthenticationWithSakRefresh_InvalidKey_Throws()
+        {
+            Action act = () => _ = new SharedAccessSignatureBuilder
+            {
+                // updating the key to change the length of it.
```

Comment:
Is the length what makes it invalid or is it because the characters/format used is incorrect?

Reply:
Its based on length
`value.Length % 4 != 0`